resources/lib/database/db_base_sqlite.py

```diff
@@ -41,18 +42,19 @@ def wrapper(*args, **kwargs):
         conn = None
         try:
             if not args[0].is_connected:
-                args[0].mutex.acquire()
+                if not G.IS_SQLITE3_THREADSAFE:
```

Comment:
This PR do not pass the CI quality gate due to too much duplicate code
you should try add a new variable for `not G.IS_SQLITE3_THREADSAFE`
like:
`is_not_thread_safe = not G.IS_SQLITE3_THREADSAFE`
then replace all occurrences
this also con cache_management py

Reply:
Fair enough.